Add documentation for Iceberg support in PrestoCPP

agrawalreetika opened the pull request:
## Description
Add documentation for Iceberg support in PrestoCPP

## Motivation and Context
Add documentation for Iceberg support in PrestoCPP

## Impact
Iceberg documentation improvement

## Test Plan
<!---Please fill in how you tested your change-->

## Contributor checklist

- [ ] Please make sure your submission complies with our [contributing guide](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md), in particular [code style](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md#code-style) and [commit standards](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md#commit-standards).
- [ ] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [ ] CI passed.

## Release Notes

```
== NO RELEASE NOTE ==
```

auden-woolfson reviewed (dismissed):
lgtm

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -308,6 +308,11 @@ the data and delete files of Iceberg tables are stored in S3. An example configu
+Presto C++ Support
+^^^^^^^^^^^^^^^^^^
+
+``HIVE``, ``NESSIE``, ``REST``, and ``HADOOP`` Iceberg catalogs are supported in PrestoCPP.
```
```suggestion
``HIVE``, ``NESSIE``, ``REST``, and ``HADOOP`` Iceberg catalogs are supported in Presto C++.
```

steveburnett commented on presto-docs/src/main/sphinx/presto-cpp.rst:
```diff
@@ -46,14 +46,21 @@ Only specific connectors are supported in the Presto C++ evaluation engine.
+
+* Iceberg tables use the Hive connector implementation in Velox.
+
+* Currently, only read operations are supported for Iceberg tables.
```
```suggestion
* Only read operations are supported for Iceberg tables.
```
Suggestion made because documentation always should describe the current state of the software.

steveburnett commented on presto-docs/src/main/sphinx/presto-cpp.rst:
```diff
@@ -46,14 +46,21 @@ Only specific connectors are supported in the Presto C++ evaluation engine.
+
+* Currently, only read operations are supported for Iceberg tables.
+
+* The Iceberg connector supports both V1 and V2 tables, including those with positional delete files, but does not currently support equality delete files.
```
```suggestion
* The Iceberg connector supports both V1 and V2 tables, including those with positional delete files, but does not support equality delete files.
```

steveburnett commented on presto-docs/src/main/sphinx/presto-cpp.rst:
```diff
@@ -46,14 +46,21 @@ Only specific connectors are supported in the Presto C++ evaluation engine.
 
-* TPCH connector, with ``tpch.naming=standard`` catalog property.
\ No newline at end of file
+For detailed information refer to :doc:`/connector/iceberg` documentation.
```
```suggestion
For more information see :doc:`/connector/iceberg`.
```

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -523,6 +528,11 @@ Property Name                                          Description
+Presto C++ Support
+~~~~~~~~~~~~~~~~~~
+
+Manifest file caching is supported in PrestoCPP.
```
```suggestion
Manifest file caching is supported in Presto C++.
```

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -546,6 +556,11 @@ JMX queries to get the metrics and verify the cache usage::
+Presto C++ Support
+~~~~~~~~~~~~~~~~~~
+
+Alluxio data caching is applicable for Presto Java. PrestoCPP supports Async data cache. See :ref:`async_data_caching_and_prefetching` for details.
```
```suggestion
Alluxio data caching is applicable for Presto Java. Presto C++ supports Async data cache. See :ref:`async_data_caching_and_prefetching` for details.
```

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -577,6 +592,11 @@ JMX queries to get the metrics and verify the cache usage::
+Presto C++ Support
+~~~~~~~~~~~~~~~~~~
+
+File and stripe footer cache is not applicable for PrestoCPP.
```
```suggestion
File and stripe footer cache is not applicable for Presto C++.
```

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -616,6 +636,11 @@ as part of a SQL query by including them in your SELECT statement.
+Presto C++ Support
+^^^^^^^^^^^^^^^^^^
+
+All above extra hidden metadata columns are supported in PrestoCPP.
```
```suggestion
All :ref:`connector/iceberg:Extra Hidden Metadata Columns` are supported in Presto C++.
```
Tested link in local doc build.

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -796,6 +821,10 @@ example uses the earliest snapshot ID: ``2423571386296047175``
+Presto C++ Support
+^^^^^^^^^^^^^^^^^^
+
+All above extra hidden metadata tables are supported in PrestoCPP.
```
```suggestion
All :ref:`connector/iceberg:Extra Hidden Metadata Tables` are supported in Presto C++.
```

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -1075,9 +1104,56 @@ Examples:
+Presto C++ Support
+^^^^^^^^^^^^^^^^^^
+
+All above procedures are supported in PrestoCPP.
```
```suggestion
All :ref:`connector/iceberg:Procedures` are supported in Presto C++.
```

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -1075,9 +1104,56 @@ Examples:
+SQL Support Summary for Presto Java and Presto CPP:
+
+============================== ============= ============ ============================================================================
+SQL Operation                  Presto Java   Presto CPP   Comments
```
```suggestion
SQL Operation                  Presto Java   Presto C++   Comments
```

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -1075,9 +1104,56 @@ Examples:
+============================== ============= ============ ============================================================================
+SQL Operation                  Presto Java   Presto CPP   Comments
+============================== ============= ============ ============================================================================
+``CREATE SCHEMA``              ✅            ✅
```
Consider changing the ✅ and ❌ in this table to the Yes and No you used in Configuration Properties and Session Properties.

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -1511,13 +1587,23 @@ schema evolution, such as adding, dropping, and renaming columns. With schema
+Presto C++ Support
+^^^^^^^^^^^^^^^^^^
+
+Schema Evolution is supported in PrestoCPP.
```
```suggestion
:ref:`connector/iceberg:Schema Evolution` is supported in Presto C++.
```

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -1511,13 +1587,23 @@ schema evolution, such as adding, dropping, and renaming columns. With schema
+Presto C++ Support
+^^^^^^^^^^^^^^^^^^
+
+PrestoCPP supports Parquet writer versions V1.
```
```suggestion
Presto C++ supports Parquet writer versions V1.
```

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -1824,6 +1910,11 @@ Query Iceberg table by specifying the tag name:
+Presto C++ Support
+^^^^^^^^^^^^^^^^^^
+
+Time Travel is supported in PrestoCPP.
```
```suggestion
Time Travel is supported in Presto C++.
```

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -1075,9 +1104,56 @@ Examples:
+
+``CREATE TABLE AS SELECT``     ✅            ❌
+
+``SELECT``                     ✅            ✅             Read is supported in PrestoCPP including those with positional delete files.
```
```suggestion
``SELECT``                     ✅            ✅             Read is supported in Presto C++ including those with positional delete files.
```

steveburnett reviewed (changes requested):
Thanks for the doc! Some nits and suggestions, nothing major.

yingsu00 commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -319,69 +324,69 @@ Configuration Properties
                                                         are ``HIVE``, ``HADOOP``, and ``NESSIE`` and ``REST``.
 
-``iceberg.hadoop.config.resources``                     The path(s) for Hadoop configuration resources.
+``iceberg.hadoop.config.resources``                     The path(s) for Hadoop configuration resources.                                                  Yes
```
The Presto C++ Support value is missing

yingsu00 commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -319,69 +324,69 @@ Configuration Properties
                                                         as a join with the data of the equality delete files.
 
-``iceberg.enable-parquet-dereference-pushdown``         Enable parquet dereference pushdown.                          ``true``
+``iceberg.enable-parquet-dereference-pushdown``         Enable parquet dereference pushdown.                          ``true``                           Yes
```
The Presto C++ Support value is missing

yingsu00 commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -319,69 +324,69 @@ Configuration Properties
-======================================================= ============================================================= ============
+======================================================= ============================================================= ================================== =================== =========================================
 
 Table Properties
```
Can you please also add a section for the C++ support for Table Properties? Just saying it's not suported because write is not implemented yet

yingsu00 commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -1511,13 +1587,23 @@ schema evolution, such as adding, dropping, and renaming columns. With schema
 evolution, users can evolve a table schema with SQL after enabling the Presto
 Iceberg connector.
 
+Presto C++ Support
```
This is for Partition Column Transform. Could you please add Presto C++ support as well? Thanks!

yingsu00 commented on presto-docs/src/main/sphinx/presto-cpp.rst:
```diff
@@ -46,14 +46,21 @@ Only specific connectors are supported in the Presto C++ evaluation engine.
+* TPCH connector, with ``tpch.naming=standard`` catalog property.
 
-* Supports reading and writing of DWRF and PARQUET file formats, supports reading ORC file format.
+Hive Connector - Iceberg Table Support
```
I think the following structure would be clearer
Supported Connectors 
- Hive connector
    - Hive table support
    - Iceberg table support
- TPCH connector
    - ....
- Fuzzer connector

yingsu00 reviewed (commented):
Thanks @agrawalreetika! I see

agrawalreetika commented:
@steveburnett @yingsu00 Thanks for your review. I have made the changes based on your comments. Please have a look at your convenience.

hantangwangd reviewed (commented):
Thanks for the doc, it's very helpful. Only a couple of nit and little thing.

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -319,69 +324,69 @@ Configuration Properties
 
-``iceberg.pushdown-filter-enabled``                     Experimental: Enable filter pushdown for Iceberg. This is     ``false``
+``iceberg.pushdown-filter-enabled``                     Experimental: Enable filter pushdown for Iceberg. This is     ``false``                          No                  Yes
                                                         only supported with Native Worker.
```
```suggestion
                                                        only supported with Presto C++.
```
Is this suggestion correct?

steveburnett commented on presto-docs/src/main/sphinx/presto-cpp.rst:
```diff
@@ -44,16 +44,32 @@ Supported Use Cases
 * Hive connector for reads and writes, including CTAS, are supported.
 
-* Hive Connector supports Delta Lake table access using Symlink table in Prestissimo.
+* Hive connector supports Delta Lake table access using Symlink table in Prestissimo.
```
```suggestion
* Hive connector supports Delta Lake table access using Symlink table in Presto C++.
```

steveburnett commented on presto-docs/src/main/sphinx/presto-cpp.rst:
```diff
@@ -44,16 +44,32 @@ Supported Use Cases
   manifests <https://docs.delta.io/latest/presto-integration.html>`_.
 
-* Iceberg tables are supported only for reads.
+* Supports reading and writing of DWRF and PARQUET file formats, supports reading ORC file format.
```
```suggestion
* Supports reading and writing of DWRF and PARQUET file formats. 

* Supports reading ORC file format.
```
Suggest breaking this out as two items for easier reading.

agrawalreetika commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -319,69 +324,69 @@ Configuration Properties
-``iceberg.max-partitions-per-writer``                   The Maximum number of partitions handled per writer.          ``100``
+``iceberg.max-partitions-per-writer``                   The Maximum number of partitions handled per writer.          ``100``                            Yes                 NA
 
-``iceberg.minimum-assigned-split-weight``               A decimal value in the range (0, 1] is used as a minimum      ``0.05``
```
I think this is because 1 is inclusive but 0 is not for this config. So it would be better to keep it exisitng way

agrawalreetika commented on presto-docs/src/main/sphinx/presto-cpp.rst:
```diff
@@ -44,16 +44,32 @@ Supported Use Cases
+
+* Only read operations are supported for Iceberg tables.
+
+* The Iceberg connector supports both V1 and V2 tables, including those with positional delete files, but does not support equality delete files.
```
So we need to create an iceberg catalog in Prestissimo as well for accessing Iceberg tables but the underline implementation on Velox is via Hive connector itself which is why I thought of keeping `Hive Connector` as main header and then different table support.

I can make the modification whatever we all think is more clear. 
@yingsu00 What's your opinion on this?

hantangwangd replied:
Sorry, I didn't describe it clearly. What I mean is, the content structure ``Hive connector/Iceberg Table Support`` is no problem, just the words ``Iceberg connector`` here may bring a little confusion. If I were a newer, I might wonder whether there is an ``Iceberg connector`` for presto C++ that I haven't noticed.

yingsu00 replied:
I think we could all it Iceberg connector because it's in Presto C++ point of view, not Velox point of view. But it would be helpful to explain that the Presto C++ Iceberg connector and catalog is backed up by Velox HiveConnector.

agrawalreetika replied:
@hantangwangd Thanks for the clarification. So here as @aditi-pandit & @yingsu00 mentioned from Presto C++ point of view it's going to be a different iceberg connector with something like this -
```
connector.name=iceberg
```
That is why I thought calling it iceberg connector would be better in the Presto C++ document.

@hantangwangd @aditi-pandit @yingsu00 Please take a look at the document, its already updated based on the points we discussed here.

hantangwangd replied:
Got the point, thank you all for your perspective @aditi-pandit @yingsu00 @agrawalreetika.

steveburnett reviewed (dismissed):
LGTM! (docs)

Pull updated branch, new local doc build, looks good. Thank you!

yingsu00 commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -319,69 +324,69 @@ Configuration Properties
                                                         as a join with the data of the equality delete files.
 
-``iceberg.enable-parquet-dereference-pushdown``         Enable parquet dereference pushdown.                          ``true``
+``iceberg.enable-parquet-dereference-pushdown``         Enable parquet dereference pushdown.                          ``true``                           Yes                 NA
```
Why is it NA? If it's fails now, just say No

yingsu00 commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -319,69 +324,69 @@ Configuration Properties
                                                         Otherwise, it will be ignored.
 
-``iceberg.file-format``                                 The storage file format for Iceberg tables. The available     ``PARQUET``
+``iceberg.file-format``                                 The storage file format for Iceberg tables. The available     ``PARQUET``                        Yes                 NA
```
Can we add a little explanation here why it's NA? We can say "NA, write is not supported yet"

yingsu00 replied:
Same for other NA items. Thanks!

yingsu00 commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -319,69 +324,69 @@ Configuration Properties
                                                         updates.
 
-``iceberg.delete-as-join-rewrite-enabled``              When enabled, equality delete row filtering is applied        ``true``
+``iceberg.delete-as-join-rewrite-enabled``              When enabled, equality delete row filtering is applied        ``true``                           Yes                 Yes
```
I don't think Presto C++ supports it. Presto C++ do not interpret the equality delete file as joins, but would compile it into domain filters or filter functions. Also, reading equality delete file PR is not merged yet. Just say No for now.

agrawalreetika replied:
Sure, I missed equality delete here.

yingsu00 commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -319,69 +324,69 @@ Configuration Properties
+                                                        only supported with Presto C++.
 
-``iceberg.rows-for-metadata-optimization-threshold``    The maximum number of partitions in an Iceberg table to       ``1000``
+``iceberg.rows-for-metadata-optimization-threshold``    The maximum number of partitions in an Iceberg table to       ``1000``                           Yes                 Yes
```
Is this one on coordinator only?

agrawalreetika replied:
As I checked in code its usage is in [IcebergMetadataOptimizer](https://github.com/prestodb/presto/blob/master/presto-iceberg/src/main/java/com/facebook/presto/iceberg/optimizer/IcebergMetadataOptimizer.java#L230) so I am assuming this is coordinator only. Do you think, I should check anything more on this?

yingsu00 commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -482,34 +487,40 @@ Deprecated Property Name                  New Property Name
+===================================================== ======================================================================= =================== ==================
+Property Name                                         Description                                                             Presto Java Support Presto C++ Support
+===================================================== ======================================================================= =================== ==================
+``iceberg.delete_as_join_rewrite_enabled``            Overrides the behavior of the connector property                        Yes                 Yes
```
Should be No, or NA

yingsu00 commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -815,6 +846,10 @@ example uses the earliest snapshot ID: ``2423571386296047175``
+Presto C++ Support
+^^^^^^^^^^^^^^^^^^
+
+All above extra hidden metadata tables are supported in Presto C++.
```
Just want to double confirm that you have verified they are working, right?

agrawalreetika replied:
Apart from`$changelog` everything works, `$changelog` throws an error which I missed adding here. I opened an issue for the same. https://github.com/prestodb/presto/issues/24816

agrawalreetika commented:
@yingsu00 I have updated all the `NA` with `No` to keep the consistency and also updated the changes based on your last few comments. Please review when ever you get a chance. Thanks!

ZacBlanco commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -482,34 +487,40 @@ Deprecated Property Name                  New Property Name
+Presto C++ Support
+^^^^^^^^^^^^^^^^^^
+
+Table properties are not supported in Presto C++ because write operations have not been implemented.
```
I think we should consider re-wording this. It's not quite clear what the statement "Table properties are not supported ..." means. Does that mean reading them? writing them? considering them during writes? I think we should be a bit more specific

yingsu00 replied:
> I think we should consider re-wording this. It's not quite clear what the statement "Table properties are not supported ..." means. Does that mean reading them? writing them? considering them during writes? I think we should be a bit more specific

I think what @ZacBlanco said makes sense. @agrawalreetika Will you be able to try them and update the doc separating the read and write

ZacBlanco commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -635,6 +661,11 @@ as part of a SQL query by including them in your SELECT statement.
+Presto C++ Support
+^^^^^^^^^^^^^^^^^^
+
+All above extra hidden metadata columns are supported in Presto C++.
```
```suggestion
All above metadata columns are supported in Presto C++.
```

> above extra hidden

Three adjectives in a row feels like a lot. I think we can reduce

agrawalreetika replied:
Here existing `extra hidden metadata columns` is taken from the main heading. But if we think we should just rephrase it saying - `All above metadata columns are supported in Presto C++` that's fine with me?

ZacBlanco replied:
It feel overly description to have three adjectives in the sentence - I think this gets the point across

ZacBlanco commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -815,6 +846,10 @@ example uses the earliest snapshot ID: ``2423571386296047175``
+Presto C++ Support
+^^^^^^^^^^^^^^^^^^
+
+All above extra hidden metadata tables, except `$changelog`, are supported in Presto C++.
```
```suggestion
All above metadata tables, except `$changelog`, are supported in Presto C++.
```

agrawalreetika replied:
Here existing `extra hidden metadata tables` is taken from the main heading. But if we think we should just rephrase it saying - `All above metadata tables, except `$changelog`, are supported in Presto C++.` that's fine with me, LMK.

ZacBlanco commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -1237,6 +1317,11 @@ Transform Name        Source Types
+Presto C++ Support
+~~~~~~~~~~~~~~~~~~
+
+Read from the tables with Partition column transform is supported in Presto C++.
```
```suggestion
Reads from tables with partition column transforms is supported in Presto C++.
```

ZacBlanco commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -1558,13 +1643,23 @@ schema evolution, such as adding, dropping, and renaming columns. With schema
+Presto C++ Support
+^^^^^^^^^^^^^^^^^^
+
+Schema Evolution is supported in Presto C++.
```
```suggestion
Schema evolution is supported in Presto C++.
```

ZacBlanco commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -1558,13 +1643,23 @@ schema evolution, such as adding, dropping, and renaming columns. With schema
+Presto C++ Support
+^^^^^^^^^^^^^^^^^^
+
+Presto C++ supports Parquet writer versions V1.
```
This is a little confusing because earlier the properties section stated that writes are not supported?

agrawalreetika replied:
Thanks for pointing it point.
It's about actually reading, so I think I can rephrase it something like - `Presto C++ supports reading Parquet data written with Parquet writer version V1.` ?

ZacBlanco commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -1871,6 +1966,11 @@ Query Iceberg table by specifying the tag name:
+Presto C++ Support
+^^^^^^^^^^^^^^^^^^
+
+Time Travel is supported in Presto C++.
```
```suggestion
Time travel queries are supported in Presto C++.
```

ZacBlanco commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -1590,13 +1670,23 @@ schema evolution, such as adding, dropping, and renaming columns. With schema
+Presto C++ Support
+^^^^^^^^^^^^^^^^^^
+
+Schema evolution is supported in Presto C++.
+
 Parquet Writer Version
 ----------------------
 
 Presto now supports Parquet writer versions V1 and V2 for the Iceberg catalog.
 It can be toggled using the session property ``parquet_writer_version`` and the config property ``hive.parquet.writer.version``.
 Valid values for these properties are ``PARQUET_1_0`` and ``PARQUET_2_0``. Default is ``PARQUET_1_0``.
 
+Presto C++ Support
+^^^^^^^^^^^^^^^^^^
+
+Presto C++ supports reading Parquet data written with Parquet writer version V1.
+
```
I see that across these `Presto C++ Support` sections we have varying forms

> ... is supported in Presto C++

and

> Presto C++ supports ....

Two comments on this

1. We should be consistent
2. I prefer that we standardize on the 2nd as it's in active voice, which is generally the form we want to write for our documentation (@steveburnett can you confirm the 2nd is the form that we should prefer?)

steveburnett replied:
Yes, we should prefer the use of active voice, but exceptions where passive voice is better can happen. Sometimes active voice can be more awkward depending on the emphasis you want to communicate. 

See 
* the Gitlab Documentation Style Guide entry for [Active Voice](https://docs.gitlab.com/development/documentation/styleguide/#active-voice) (shorter)
* the Google Developer Documentation Style Guide entry for [Active Voice](https://developers.google.com/style/voice) (longer and more examples)

tl;dr: use active voice, but if active voice is awkward then passive voice is okay.

steveburnett replied:
In this context of the "Presto C++ Support" subtopics, I would argue that passive voice is better. 

* The statements immediately follow the heading "Presto C++ Support" so "Presto C++ supports" repeat the information without adding to it, and 

* Passive voice here follows the guideline in the Google Developer Documentation Style Guide entry for [Active Voice](https://developers.google.com/style/voice) "To emphasize an object over an action." 

Using passive voice here emphasizes the object "HIVE, NESSIE, REST, and HADOOP Iceberg catalogs" and makes it easier for the user to find the information they need. 

See the screenshots I made in a local doc build to show the difference. I think the second one is the better one. 

1)
<img width="610" alt="Screenshot 2025-04-08 at 10 20 42 AM" src="https://github.com/user-attachments/assets/7d05aa19-06a0-4de6-8a65-ca10c839b56a" />

2)
<img width="688" alt="Screenshot 2025-04-08 at 10 20 15 AM" src="https://github.com/user-attachments/assets/49d0aac8-cd20-4406-86ae-2acce678a090" />

ZacBlanco reviewed (changes requested):
One comment on consistency in our grammar, otherwise I think this looks great. Thanks!

steveburnett commented on presto-docs/src/main/sphinx/connector/iceberg.rst:
```diff
@@ -470,46 +475,47 @@ Some table properties have been deprecated or removed. The following table lists
+===================================================== ======================================================================= =================== =============================================
+Property Name                                         Description                                                             Presto Java Support Presto C++ Support
+===================================================== ======================================================================= =================== =============================================
+``iceberg.delete_as_join_rewrite_enabled``            Overrides the behavior of the connector property                        Yes                 No, equality delete read is not supported yet
```
```suggestion
``iceberg.delete_as_join_rewrite_enabled``            Overrides the behavior of the connector property                        Yes                 Equality delete read is not supported
```

agrawalreetika replied:
updated

steveburnett reviewed (changes requested):
Thanks for the change and the new consistency! One nit.

ZacBlanco reviewed (approved):
Thanks for the changes!

steveburnett reviewed (approved):
LGTM! (docs)

Pull updated branch, new local doc build, looks good. Thanks!
